### DISCUSSION TITLE
Live dictation combined with KaldiAG voice coding for computer control

### DISCUSSION THREAD

**Discussion by shervinemami** [openai/whisper#285] [Show and tell]
I got OpenAI Whisper working as a natural language dictation backend for the "Kaldi Active Grammar" speech recognition / voice coding framework, allowing me to control my computer in real-time (using commands in Kaldi) with the ability to also write plain text using Whisper for dictation. Hence able to get by without a keyboard, for web browsing, document writing and coding / programming.

My code is just hacked together for now but it's a useable start, I'm sure the Kaldi Active Grammar folks will integrate it better soon.

Demo video: https://youtu.be/7OCMu6JRL-8

(Instructions for getting this working are on the KaldiAG Gitter chat at "https://gitter.im/kaldi-active-grammar/community?at=6342b59d773e7e5c7bc87cb2")

**Comment by DanRioDev** [comment]
How do you got Whisper_mic to work, that piece just gives me `ALSA` errors it won't start no matter troubleshooting I follow from the internet.
